applications/dot-ring.md

```diff
@@ -0,0 +1,258 @@
+    -   Jubjub (and Banderwagon for use with Bandersnatch proofs if applicable)
+    -   Ed25519 (EdDSA)
+    -   NIST P-256
+    -   (Future considerations, if time/resources permit beyond core scope: Pallas, other NIST curves P-384/P-521, secp256k1, BLS12-381)
+-   **Standard Cryptographic Functions**: Key generation, point compression/decompression, ECDSA/EdDSA signing and verification (where applicable for the curve).
+-   **Hashing to Elliptic Curves (RFC 9380)**:
+    -   Full implementation of all 14 suites listed in Section8 of RFC 9380 ("Suites for Hashing") for the implemented curves. This includes:
+        -   Suites for P-256 (e.g., P256_XMD:SHA-256_SSWU_RO_, P256_XMD:SHA-256_SSWU_NU_)
+        -   Suites for edwards25519 (e.g., edwards25519_XMD:SHA-512_ELL2_RO_, edwards25519_XOF:SHAKE256_ELL2_RO_)
+        -   Support for generic modes like `ExpandMsgXOF` (e.g., with SHAKE256) and `ExpandMsgXMD`.
+        -   Implementation of underlying mapping-to-curve algorithms (Shallue-van de Woestijne, Simplified SWU, Elligator 2, etc.) as required by the suites.
+-   Thorough unit tests (aiming for >80% coverage), extensive API documentation, and clear usage examples.
+-   Initial benchmarks for all core operations on supported curves.
+
+References: [Masson, Sanso, and Zhang's work on Bandersnatch (ePrint 2021/1152)](https://eprint.iacr.org/2021/1152.pdf), [RFC 9380 - Hashing to Elliptic Curves](https://www.rfc-editor.org/rfc/rfc9380.html).
+
+##### 2. `pyvrf`: Verifiable Random Function Library
+
+Objective: To implement IETF standard VRFs and other significant VRF variants (Pedersen, Ring), leveraging `ec-crypto` for underlying curve operations.
+Deliverables:
+-   **IETF ECVRF (RFC 9381)**:
+    -   Full implementation of ECVRF algorithms (Section5 & Section6 of RFC 9381) for P-256 and Ed25519, supporting relevant ciphersuites (e.g., ECVRF-P256-SHA256-TAI, ECVRF-EDWARDS25519-SHA512-TAI, ECVRF-P256-SHA256-SSWU, etc.).
+    -   Correct handling of domain separation tags (DSTs) and all helper functions as per the RFC.
+    -   Verification against official RFC test vectors.
+-   **Pedersen VRF**: Implementation for Bandersnatch.
+-   **Ring VRF**: The core Bandersnatch Ring VRF scheme, integrating with `dotring` (for ring proofs) and `fiat-shamir`.
+-   Modular design for potential future additions of VRF schemes or curve support.
+-   Comprehensive testing, API documentation, and examples.
+
+References: [RFC 9381 - Verifiable Random Functions (VRFs)](https://datatracker.ietf.org/doc/html/rfc9381), [W3F Bandersnatch-VRF specification](https://github.com/davxy/bandersnatch-vrf-spec).
+
+##### 3. `fiat-shamir`: Generic Transcript Layer
+
+Objective: To provide a robust, secure, and flexible transcript mechanism using extendable-output functions (XOFs) for building non-interactive proof systems, following CFRG guidelines.
+Deliverables:
+-   Secure transcript abstraction supporting SHAKE128 and SHAKE256 as XOFs.
```

Comment:
We use ark-transcript so maybe here mention that you would imlpmenet a version fully compatible with ark-transcript and refer to ark-transcript crate.

Reply:
Added the mentions